### PR TITLE
Docs: Create examples section and remove some outdated pages (not all).

### PR DESCRIPTION
**Pull request by dhatch**

### CODE REVIEW

File: docs/content/any/examples/gitclub.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+title: GitClub
+description: "GitClub is a complete example application with backends
+using Oso for authorization in multiple languages."
+href: https://github.com/osohq/oso
```
Comment (gj) [inline docs/content/any/examples/gitclub.md:5]: What's the goal here? Should this point to the GitClub repo (maybe for the currently selected language)?

File: docs/content/any/examples/inheritance/index.md [hunk 2]
```diff
@@ -4,6 +4,7 @@ docname: using/examples/inheritance
 images: {}
 path: /using-examples-inheritance
 title: Share Rules Across Resources
+weight: 10
 description: |
   Oso policies make it possible to share rules across related resource types, and override them as needed.
 aliases:
```
Comment (gj) [inline docs/content/any/examples/inheritance/index.md:10]: probably need to add a new alias, no?

**Review by gj — CHANGES_REQUESTED** [review]
main question is: should we add aliases for all the pages we're moving/deleting?

File: docs/content/any/examples/gitclub.md (hunk 1 above)
Comment (dhatch) [reply docs/content/any/examples/gitclub.md:5]: Yeah this should point to gitclub, thanks for catching.

File: docs/content/any/examples/_index.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+title: Examples
+weight: 3
+isAny: true
+---
```
Comment (gj) [inline docs/content/any/examples/_index.md:5]: should we add an alias for the deleted `docs/content/any/guides/more/_index.md` index here?

**Review by gj — CHANGES_REQUESTED** [review]
@dhatch did you test out the aliases locally? Not totally sure what's going on, but navigating to http://localhost:1313/guides/more/inheritance.html (no lang specified) takes me to http://localhost:1313/ruby/examples/inheritance.html (ruby)

and navigating to http://localhost:1313/guides/more/context.html (no lang specified) takes me to http://localhost:1313/java/examples/context.html (java)

**Comment by dhatch** [comment]
> @dhatch did you test out the aliases locally? Not totally sure what's going on, but navigating to http://localhost:1313/guides/more/inheritance.html (no lang specified) takes me to http://localhost:1313/ruby/examples/inheritance.html (ruby)
> 
> and navigating to http://localhost:1313/guides/more/context.html (no lang specified) takes me to http://localhost:1313/java/examples/context.html (java)

Yeah, I believe this is just an issue with how aliases work in the setup we have. I don't think there's a fix, can confirm with @samscott89

File: docs/content/any/examples/_index.md (hunk 3 above)
Comment (dhatch) [reply docs/content/any/examples/_index.md:5]: No it's very unlikely there are any links to that page.

**Review by samscott89 — APPROVED** [review]
Sorry that Hugo aliases are bad.

**Pull request merged**